### PR TITLE
[CSL-2863] Proactive pii detection

### PR DESCRIPTION
**Pull request by stanlp1**
Since all of our tracking requests go through the request queue, added logic in the request queue to not add a request to queue if PII is detected. Only the path, query parameters are checked at the moment.

### CODE REVIEW

**Review by esezen — APPROVED** [review]
LGTM! Diffend isn't happy about some packages but they don't seem related to this PR. We'll handle that separately

**Comment by esezen** [comment]
@jjl014 do you want to take another look or am I good to merge & deploy?

File: src/utils/request-queue.js [hunk 1]
```diff
@@ -33,6 +34,10 @@ class RequestQueue {
     if (this.sendTrackingEvents && !this.humanity.isBot()) {
       const queue = RequestQueue.get();
 
+      if (requestContainsPii(url, body)) {
+        return;
+      }
```
Comment (jjl014) [inline src/utils/request-queue.js:39]: We could probably move this right before line 35 where we get the queue.

**Review by jjl014 — APPROVED** [review]
Left one super minor comment, but this looks great to me! Shippit! 🚀 🚢

File: src/utils/request-queue.js (hunk 1 above)
Comment (esezen) [reply src/utils/request-queue.js:39]: 🧠

Comment (stanlp1) [reply src/utils/request-queue.js:37]: 👍

**Pull request merged**